mimic/model/flavor_objects.py

```diff
@@ -1,279 +1,107 @@
     """
-    flavors = {"15 GB I/O v1": {"id": "io1-15", "ram": 15360, "vcpus": 4, "rxtx_factor": 1250,
-                                "disk": 40},
-               "30 GB I/O v1": {"id": "io1-30", "ram": 30720, "vcpus": 8, "rxtx_factor": 2500,
-                                "disk": 40},
-               "60 GB I/O v1": {"id": "io1-60", "ram": 61440, "vcpus": 16, "rxtx_factor": 5000,
-                                "disk": 40},
-               "90 GB I/O v1": {"id": "io1-90", "ram": 92160, "vcpus": 24, "rxtx_factor": 7500,
-                                "disk": 40},
-               "120 GB I/O v1": {"id": "io1-120", "ram": 122880, "vcpus": 32, "rxtx_factor": 10000,
-                                 "disk": 40}}
-
-    def extra_specs_json(self):
+    def flavor_by_id(self, flavor_id):
         """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for a memory flavor.
+        Retrieve a :obj:`Flavor` object by its ID.
         """
-        return {
-            "class": "io1",
-            "policy_class": "io_flavor"
-        }
-
-
-class RackspaceMemoryFlavor(Flavor):
-    """
-    A Rackspace memory flavor object representation
-    """
-    flavors = {"15 GB Memory v1": {"id": "memory1-15", "ram": 15360, "vcpus": 2, "rxtx_factor": 625,
-                                   "disk": 0},
-               "30 GB Memory v1": {"id": "memory1-30", "ram": 30720, "vcpus": 4, "rxtx_factor": 1250,
-                                   "disk": 0},
-               "60 GB Memory v1": {"id": "memory1-60", "ram": 61440, "vcpus": 8, "rxtx_factor": 2500,
-                                   "disk": 0},
-               "120 GB Memory v1": {"id": "memory1-120", "ram": 122880, "vcpus": 16, "rxtx_factor": 5000,
-                                    "disk": 0},
-               "240 GB Memory v1": {"id": "memory1-240", "ram": 245760, "vcpus": 32,
-                                    "rxtx_factor": 10000, "disk": 0}}
+        if len(self.flavors_store) < 1:
+            self.create_flavors_list()
+        for flavor in self.flavors_store:
+            if flavor.flavor_id == flavor_id:
+                return flavor
 
-    def extra_specs_json(self):
+    def create_flavors_list(self):
         """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for a flavor.
+        Generates the data for each flavor in each flavor class
         """
-        return {
-            "class": "memory1",
-            "policy_class": "memory_flavor"
-        }
+        rackspace_flavors = [RackspaceStandardFlavor, RackspaceComputeFlavor,
+                             RackspacePerformance1Flavor, RackspaceOnMetalFlavor,
+                             RackspacePerformance2Flavor, RackspaceMemoryFlavor,
+                             RackspaceIOFlavor, RackspaceGeneralFlavor]
 
+        if len(self.flavors_store) < 1:
```

Comment:
If you check `len(self.flavors_store) < 1` in this function's callers, then you don't need to check it here. Or you could check it here, drop the other checks and call it `_create_flavors_list_if_empty` or similar. I feel checking it in both places is unnecessary repetition.